renpy/display/image.py

```diff
@@ -937,9 +934,8 @@ def apply_attributes(self, layer, tag, name, wanted=[], remove=[]):
             if f is not None:
                 defaults = f(name)
 
-        # Add any defaults to the set of attributes a matching image may have.
-        if defaults is not None:
-            optional.extend(defaults)
+        # The list of attributes a matching image may have.
+        optional = list(defaults) if defaults else [ ]
```

Comment:
Maybe keeping this a None-check instead of a -check ? Like `[] if defaults is None else list(defaults)`

Reply:
This was done intentionally to avoid passing empty iterables to the list constructor. It's quite common for `()` to show up here when the callback is not being used (any image without active attributes will get an empty tuple), and the creator could also choose to return `set()`, `[]`, or `()` from the callback, which seems reasonable to me if they're already working with those data types in their function.

Conversely, I can't think of any valid scenario in which we'd want to send a falsey value to the list constructor over using the more optimised empty list literal. The worst possible outcome I can think of at the moment would be that non-iterable falsey values would be swallowed rather than triggering a stack trace, which in this scenario feels acceptable to avoid calling the list constructor unnecessarily for the vast majority of show statements.